Refactor of contribute pages

bedroesb opened the pull request:
This PR includes the wording of the contribute pages for the three different ways available.
It also includes correction of typos in other pages of this section.

Preview [here](https://elixir-belgium.github.io/rdmkit-sandbox/how_to_contribute)

bianchini88 commented on pages/contribute/editors_checklist.md:
```diff
@@ -19,9 +19,9 @@ summary: Checklist for editors before approving and merging a pull request (PR).
 10. When a new page is added, a news item is added to the [news.yml file](https://github.com/elixir-europe/rdmkit/blob/master/_data/news.yml), in the same branch of the PR.
-11. The contributors are thanked for their effort and informed about the publication of their content.
-12. The PR is linked to related issues and can be merged in main branch with no conflicts.
+11. The contributors are thanked for their efforts and informed about the publication of their content.
```
```suggestion
11. The contributors are acknowledged for their efforts and informed about the publication of their content.
```

bianchini88 commented on pages/contribute/github_way.md:
```diff
@@ -1,70 +1,77 @@
 
+Prerequisites:
+* No technical knowledge
+* A GitHub account. If you don’t have one, [create a free GitHub account](https://github.com/join) before you start.
```
```suggestion
* A GitHub account. If you do not have one, [create a free GitHub account](https://github.com/join) before you start.
```

bianchini88 commented on pages/contribute/google_doc_way.md:
```diff
@@ -1,24 +1,20 @@
-* The editors will transfer your page to GitHub and let you know when it is published on the website.
+3. Notify the editorial team when you are finished.
+4. Read feedback. The editors will assign reviewers to your page, who will provide feedback as comments on the shared Google Doc.
+5. Address the reviewers’ comments and let the editorial team know that you’ve finished reviewing.
```
```suggestion
5. Address the reviewers’ comments and let the editorial team know when you are finished.
```

bianchini88 commented on pages/contribute/how_to_contribute.md:
```diff
@@ -1,39 +1,30 @@
+## Acknowledgement and ownership of content
 
-## Ownership of content
+Contributors will be shown at the bottom of each page and on the main [contributors page](https://rdmkit.elixir-europe.org/contributors). if listed in the metadata of the markdown file. All contributors will also be displayed on the. If you want to link your GitHub account, ORCID or email address, please add your name and corresponding information to the [CONTRIBUTORS file](https://github.com/eli ...
```
> All contributors will also be displayed on the.

This sentence does not seem to be complete

martin-nc replied:
I'm guessing that's a fragment that shouldn't be there. Maybe the passage should read:

> Contributors will be shown at the bottom of each page and on the main [contributors page](https://rdmkit.elixir-europe.org/contributors) if listed in the metadata of the markdown file. If you want to link your GitHub account, ORCID or email address, please add your name and corresponding information to the [CONTRIBUTORS file](https://github.com/elixir-europe/rdmkit/blob/master/_data/CONTRIBUTORS.yaml).

bianchini88 replied:
> I'm guessing that's a fragment that shouldn't be there.

I think you are right.

> If you want to link your GitHub account, ORCID or email address, please add your name and corresponding information to the [CONTRIBUTORS file](https://github.com/elixir-europe/rdmkit/blob/master/_data/CONTRIBUTORS.yaml).

I never realised adding your information to the CONTRIBUTORS yaml was not mandatory. I think we should require some information (at least the ORCID) since we are promoting best practices. Any thoughts @bedroesb?

bedroesb replied:
Can we assume everyone has a ORCID?

bianchini88 replied:
> Can we assume everyone has an ORCID?

It would be my assumption, but maybe I am living inside a bubble. ORCID is typically required by scientific journals to publish, so at least there is a strong push for scientists to create one anyway. (I created mine 7 years ago)

bianchini88 commented on pages/contribute/tool_resource_update.md:
```diff
@@ -7,11 +7,11 @@ summary: How to add a tool or resource to RDMkit
 The [all_tools_and_resources](all_tools_and_resources) list is based on the [yaml file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml) in the `_data` directory of the RDMkit repository. Tools and resources can be manually linked to [FAIRsharing.org](https://fairsharing.org/), [Bio.tools](https://bio.tools) and [TeSS](https://tess.elixir-europe.org/), but eve ...
 
-{% include callout.html type="important" content="The link with FAIRsharing,TeSS and Bio.tools is automatically done using GitHub actions and is weekly updated. If no FAIRsharing ID, Bio.tools ID or TeSS Query is available for a source, but there is yet one automatically given (faulty), you can overwrite the automatic linking by adding 'NA' as registry." %}
+{% include callout.html type="important" content="The link with FAIRsharing,TeSS and Bio.tools is automatically done using GitHub actions and is updated weekly. If no FAIRsharing ID, Bio.tools ID or TeSS Query is available for a source, but there is yet one automatically given (faulty), you can overwrite the automatic linking by adding 'NA' as a registry." %}
```
```suggestion
{% include callout.html type="important" content="The link with FAIRsharing,TeSS and Bio.tools is automatically done using GitHub actions and is updated weekly. If no FAIRsharing ID, Bio.tools ID or TeSS query is available for a source, but there is yet one automatically given (faulty), you can overwrite the automatic linking by adding 'NA' as a registry." %}
```

bianchini88 commented on pages/contribute/tool_resource_update.md:
```diff
@@ -20,7 +20,7 @@ Each tool or resource mentioned in the text has metadata stored in the [main yam
 - **url**: URL to the main page of the tool or resource, make sure to let the URL start with `https://`
 - **description**: A short description of the tool or resource. Try to not use the characters `"` or `'` 
-- **registry**: 3 registries are supported: [Bio.tools](https://bio.tools), [FAIRsharing.org](https://fairsharing.org/) and [TeSS](https://tess.elixir-europe.org/). The keywords you can use respectively are: `biotools`, `fairsharing`, `fairsharing-coll` and `tess`, specifying the id or query with a colon. FAIRsharing collections have an ID that follows the pattern `bsg-s000XXX`. List registries u ...
+- **registry**: 3 registries are supported: [Bio.tools](https://bio.tools), [FAIRsharing.org](https://fairsharing.org/) and [TeSS](https://tess.elixir-europe.org/). The keywords you can use respectively are: `biotools`, `fairsharing`, `fairsharing-coll` and `tess`, specifying the id or query with a colon. FAIRsharing collections have an ID that follows the pattern `bsg-s000XXX`. List registries u ...
```
```suggestion
- **registry**: 3 registries are supported: [Bio.tools](https://bio.tools), [FAIRsharing.org](https://fairsharing.org/) and [TeSS](https://tess.elixir-europe.org/). The keywords you can use respectively are: `biotools`, `fairsharing`, `fairsharing-coll` and `tess`, specifying the id or query with a colon. FAIRsharing collections have an ID that follows the pattern `bsg-s000XXX`. List registries under the `registry` attribute as `key: value pairs`. If no FAIRsharing ID, Bio.tools ID or TeSS query is available for a source, you can overwrite the automatic linking by adding 'NA' as a registry.
```

martin-nc commented on pages/contribute/editors_checklist.md:
```diff
@@ -19,9 +19,9 @@ summary: Checklist for editors before approving and merging a pull request (PR).
 5. Make sure that listed tools or resources are tagged in the text with the [correct snippet](tool_resource_update) + that its metadata is described in the [tool_and_resource_list.yml](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml) file. 
 6. Check if relevant recipes on FAIR Cookbook can be linked to the RDMkit page that is being added/changed ([Linking from RDMkit to FAIR Cookbook](editorial_board_guide#linking-from-rdmkit-to-fair-cookbook)).
-7. The content is conform to RDMkit scope, [style](style_guide) and templates.
+7. The content conforms to RDMkit scope, [style](style_guide) and templates.
```
> The content conforms to RDMkit scope, ...

The content conforms to the RDMkit's scope,...

I think it needs a possessive.

bianchini88 replied:
```suggestion
7. The content conforms to RDMkit's scope, [style](style_guide) and templates.
```

martin-nc commented on pages/contribute/github_way.md:
```diff
@@ -1,70 +1,77 @@
-1. Editors publish your content.
+2. Click “Issues” in the top menu bar and look at the existing issues. See if your idea or suggestion is already being discussed.
+    * If an issue exists, comment and let people know if you want to contribute.
+    * If no relevant issue exists, create a new issue by clicking the green “New issue” button on the right and choosing one of the issue templates. You can find more information on creating issues in the [GitHub documentation](https://docs.github.com/en/github/managing-your-work-on-github/creating-an-issue).
```
> ...and choosing one of the issue templates.

...and choose one of the issue templates.

"Choose" needs to be imperative, I think, like "create" before it.

bianchini88 replied:
I think the intended coordination is between "clicking" and "choosing", not between "create" and "choosing" (create by clicking and choosing).

bedroesb replied:
@martin-nc do you agree here with @bianchini88 ?

martin-nc replied:
Oh, by the time I'd got to "choosing" I'd forgotten about "create"! I suppose you could say:

> If no relevant issue exists, create a new issue: click the green “New issue” button on the right and choose one of the issue templates.

Or just leave it as it is.

bedroesb replied:
Sounds good to me!

martin-nc commented on pages/contribute/github_way.md:
```diff
@@ -1,70 +1,77 @@
+Before starting editing on GitHub:
+1. Make sure you are following our [style guide](https://rdmkit.elixir-europe.org/style_guide).
+2. Follow the structure of the provided template for the page you wish to create or update. 
+3. We use markdown. To know how to create paragraphs, headings, format text, add links and images and much more, follow our [markdown cheat sheet](https://rdmkit.elixir-europe.org/markdown_cheat_sheet).
```
> To know how...

To learn how...

bianchini88 replied:
```suggestion
3. We use markdown. To learn how to create paragraphs, headings, format text, add links and images and much more, follow our [markdown cheat sheet](https://rdmkit.elixir-europe.org/markdown_cheat_sheet).
```

martin-nc commented on pages/contribute/google_doc_way.md:
```diff
@@ -1,24 +1,20 @@
-* Address the reviewers' comments and let the editorial team know, again by mail, that you're finished with revisions.
-* The editors will transfer your page to GitHub and let you know when it is published on the website.
+3. Notify the editorial team when you are finished.
+4. Read feedback. The editors will assign reviewers to your page, who will provide feedback as comments on the shared Google Doc.
```
> Read feedback.

Read the feedback.

bianchini88 replied:
```suggestion
4. Read the feedback that the reviewers will provide as comments on the shared Google document.
```

martin-nc commented on pages/contribute/how_to_contribute.md:
```diff
@@ -1,39 +1,30 @@
+This project is only possible with the many [community contributors](https://rdmkit.elixir-europe.org/contributors). RDMkit is an open project where everyone can contribute to the site.
 
-Everyone is welcome to contribute to this site, and we've tried to make it as easy as possible. Choose a way that suits you. The website is hosted on GitHub, so:
+If you wish to contribute to this community-led project, we ask you to remember the following for better consistency and efficiency: 
```
> If you wish to contribute to this community-led project, we ask you to remember the following for better consistency and efficiency:

If you wish to contribute, please:

Maybe we can keep this shorter, and not repeat that it is a community project.

bianchini88 replied:
```suggestion
If you wish to contribute, please remember the following for better consistency and efficiency: 
```

martin-nc commented on pages/contribute/how_to_contribute.md:
```diff
@@ -1,39 +1,30 @@
-* Please read our [Markdown cheat sheet](markdown_cheat_sheet) in order to be able to fill in the page metadata and to format the layout correctly.
-* To avoid having to resolve many issues during the revision process, please check the [Editors checklist](editors_checklist) and make sure you have complied with the requirements.
-* If others were involved in your contribution, by writing up or by providing resources such as diagrams or links, please make sure you acknowledge them in the contributors' section of your page, after receiving their permission.
+* Google Doc. If you are unfamiliar with Git and prefer to use a classic document editor, follow [the Google Doc way](https://rdmkit.elixir-europe.org/google_doc_way)
```
> ...and prefer to use a classic document editor,...

...and prefer to use a Word-like editor,... ?

Wasn't sure what a "classical" document editor was. QuarkXPress is also a document editor with a long history, but nothing like Google Docs.

bedroesb replied:
I agree with docs and started changing changed it further at some places accordingly

martin-nc commented on pages/contribute/tool_resource_update.md:
```diff
@@ -43,11 +43,11 @@ Tools and resources specifically mentioned in the text of the pages should be pr
 1. Make sure the tool you want to add is not yet already described in the [yaml file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml). If so, go to step 3, if not, go follow the next step.
 
-1. Click on the pencil icon seen on GitHub of the [main yaml file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml) as described in our GitHub Guide. Add your tool or resource at the bottom of the file following the structure described in the [The main yaml file section of this page](#the-main-yaml-file). Make sure the indentation follows the one of the previo ...
+1. Click on the pencil icon seen on GitHub of the [main yaml file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml) as described in our GitHub Guide. Add your tool or resource at the bottom of the file following the structure described in the [The main yaml file section of this page](#the-main-yaml-file). Make sure the indentation follows one of the previously ...
```
> ...in case of doubt.

...to check the format.

bianchini88 replied:
```suggestion
1. Click on the pencil icon seen on GitHub of the [main yaml file](https://github.com/elixir-europe/rdmkit/blob/master/_data/tool_and_resource_list.yml) as described in our GitHub Guide. Add your tool or resource at the bottom of the file following the structure described in the [The main yaml file section of this page](#the-main-yaml-file). Make sure the indentation follows one of the previously listed items. In case of doubt, copy the content of the yaml file and paste it into an online yaml validator to check the format.
```

martin-nc commented on pages/contribute/ways_to_contribute.md:
```diff
@@ -0,0 +1,3 @@
+Google Doc. If you are unfamiliar with Git and prefer to use a classic document editor, follow the Google Doc way
```
> ...classic document editor...

...Word-like editor...? See my comment above.

martin-nc reviewed (commented):
Just a few suggestions to take or leave!

bianchini88 commented on pages/contribute/google_doc_way.md:
```diff
@@ -1,24 +1,20 @@
 ---
-title: Google doc way
+title: Google Doc way
```
```suggestion
title: Google Docs way
```

bianchini88 commented on pages/contribute/how_to_contribute.md:
```diff
@@ -1,39 +1,30 @@
-* Please read our [Markdown cheat sheet](markdown_cheat_sheet) in order to be able to fill in the page metadata and to format the layout correctly.
-* To avoid having to resolve many issues during the revision process, please check the [Editors checklist](editors_checklist) and make sure you have complied with the requirements.
-* If others were involved in your contribution, by writing up or by providing resources such as diagrams or links, please make sure you acknowledge them in the contributors' section of your page, after receiving their permission.
+* Google Doc. If you are unfamiliar with Git and prefer to use a classic document editor, follow [the Google Doc way](https://rdmkit.elixir-europe.org/google_doc_way)
```
```suggestion
* Google Docs. If you are unfamiliar with Git and prefer to use a classic document editor, follow [the Google Docs way](https://rdmkit.elixir-europe.org/google_doc_way)
```

bianchini88 commented on pages/contribute/how_to_contribute.md:
```diff
@@ -1,39 +1,30 @@
+## Acknowledgement and ownership of content
 
-## Ownership of content
+Contributors will be shown at the bottom of each page and on the main [contributors page](https://rdmkit.elixir-europe.org/contributors). if listed in the metadata of the markdown file. All contributors will also be displayed on the. If you want to link your GitHub account, ORCID or email address, please add your name and corresponding information to the [CONTRIBUTORS file](https://github.com/eli ...
```
```suggestion
Contributors will be shown at the bottom of each page and on the main [contributors page](https://rdmkit.elixir-europe.org/contributors) if listed in the metadata of the markdown file. If you want to link your GitHub account, ORCID or email address, please add your name and corresponding information to the [CONTRIBUTORS file](https://github.com/elixir-europe/rdmkit/blob/master/_data/CONTRIBUTORS.yaml).
```

bianchini88 commented:
@bedroesb, I incorporated the feedback from @martin-nc as "suggested changes". Some things still need to be discussed, though.

bedroesb commented:
@bianchini88 and @martin-nc thanks a lot for the feedback and commits, I've implemented some of them, and I also changed some things, including something that surfaced a while ago at the drawing board:

![image](https://github.com/user-attachments/assets/553685ba-9c97-437a-b96d-92962086b38d)

Tiles

bedroesb commented:
You can preview all the changes in this pull request here: https://elixir-belgium.github.io/rdmkit-sandbox/how_to_contribute

bianchini88 commented:
@bedroesb I liked the clickable boxes quite a lot. Although I am not sure we need all the text inside of them... the "if you are unfamiliar with Git" is not very useful, as it is very clear from the graphics that GitHub and Google Docs are alternative approaches.

bianchini88 commented on pages/contribute/how_to_contribute.md:
```diff
@@ -2,38 +2,60 @@
+        <a href="{{ 'github_way' | relative_url }}" class="stretched-link">
+          <h3 class="card-title text-dark mt-0">GitHub</h3>
+        </a>
+        <p class="card-text">If you are unfamiliar with Git but want to try contributing on GitHub with no technical knowledge required.</p>
```
```suggestion
        <p class="card-text">If you are unfamiliar with Git but want to contribute on GitHub. No technical knowledge is required.</p>
```

bedroesb commented:
@bianchini88 I agree with you, what do you think of this sentence:

![image](https://github.com/user-attachments/assets/94e624d0-e26e-488f-969b-474d09e4ccf6)

bianchini88 commented:
@bedroesb It looks much better. One could try to shorten the first one to make it fit 2 lines, e.g. "Contribute to the content directly. No technical knowledge is required."

martin-nc commented:
Yes, or "Add or edit content on GitHub. No technical knowledge required". I'm not sure if people will know what "directly" means here, and they might think it means editing the text of the live site, which is a bit scary! 

I also wonder if we can say "No technical knowledge required", since you do have to know markdown. It's not a high bar, but on the [GitHub contribute page](https://rdmkit.elixir-europe.org/github_way) we link to [GitHub's own guides](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github). There are three guides here, which is quite a bit to go though if you just want to add a paragraph or two! Maybe it would be less overwhelming to remove that link and just leave the link to our own markdown cheatsheet. I'm guessing that's all most people will need.

bianchini88 commented on pages/contribute/github_way.md:
```diff
@@ -3,68 +3,74 @@ title: GitHub way
-1. Editors publish your content.
+2. Click “Issues” in the top menu bar and look at the existing issues. See if your idea or suggestion is already being discussed.
+    * If an issue exists, comment and let people know if you want to contribute.
+    * If no relevant issue exists, create a new issue by clicking the green “New issue” button on the right and choosing one of the issue templates. You can find more information on creating issues in the [GitHub documentation](https://docs.github.com/en/github/managing-your-work-on-github/creating-an-issue).
```
```suggestion
    * If no relevant issue exists, create a new one: click the green “New issue” button on the right and choose a template. You can find more information on creating issues in the [GitHub documentation](https://docs.github.com/en/github/managing-your-work-on-github/creating-an-issue).
```

bianchini88 replied:
I implemented @martin-nc's suggestion above, as it is clearer. I made additional modifications to avoid repeating the word "issue" too often.

bedroesb replied:
Woops didn't see this comment, I've also implemented it !

bedroesb commented:
@martin-nc and @bianchini88 what do you think about the fact that we are leaving out the current screenshots? 
![image](https://github.com/user-attachments/assets/c03013d7-1a94-412b-afed-9e683ce792ed)

bedroesb commented:
Reminder that everything can be previewed here: https://elixir-belgium.github.io/rdmkit-sandbox

bianchini88 commented:
@bedroesb @martin-nc One could argue that md syntax is actually a prerequisite and we could link the style guide there. I do not quite like the "Prerequisites: No technical knowledge" bit. It feels like we are putting a bit too much emphasis on this.

The style guide is linked already at the very beginning of https://elixir-belgium.github.io/rdmkit-sandbox/how_to_contribute, so it is not strictly necessary to link it again in https://elixir-belgium.github.io/rdmkit-sandbox/github_way (although it probably does not do much harm)

bedroesb commented:
> @bedroesb if I understand correctly, you are asking for a comparison between these two pages:
> 
> 1. https://elixir-belgium.github.io/rdmkit-sandbox/github_way
> 2. https://rdmkit.elixir-europe.org/github_way
> 
> Right?

Yes!

martin-nc commented:
Yes, I think @bianchini88 is right about the prerequisite bit. We don't need to mention a negative prerequisite. I mean, you can still use the GitHub method even if you _do_ have technical knowledge! I also think a little knowledge of markdown is a prerequisite, and a link to the cheatsheet might be handy here. So the bullet point could be "Basic knowledge of markdown. All you need to know is in our [markdown cheatsheet](https://rdmkit.elixir-europe.org/markdown_cheat_sheet)".

As for screenshots, I've found that most people like them in instructions (and so do I!). They just confirm you are looking in the right place.

In the new "[Start editing](https://elixir-belgium.github.io/rdmkit-sandbox/github_way#start-editing)" section, it says:

> If you want to contribute to an existing page, go to the page on the site and click the “Edit me” pencil icon

I think we might need to say where the icon is, like "...next to the page title".

bianchini88 commented:
@bedroesb I agree with everything @martin-nc is stating above. I am also wondering whether it is the case to add a link to https://rdmkit.elixir-europe.org/page_metadata#possible-metadata-attributes-of-a-page.

bedroesb commented:
After a last round of feedback I:
- Linked towards the page metadata guide
- Brought back the screenshots
- Updated the Prerequisites including MD guide
- Left out the Suggestion form since we have it in contact

bedroesb commented:
@bianchini88 and @martin-nc for completeness!

bianchini88 commented:
At the top of [this page](https://elixir-belgium.github.io/rdmkit-sandbox/how_to_contribute) what do we mean exactly with 

> Provide acknowledgement for all your contributions.

?

bianchini88 commented:
> If you want to link your GitHub account, ORCID or email address,

We have never discussed whether adding this information should be mandatory or not. I would argue that all people have an institutional email address and most people have an ORCID, as this information is also required for e.g. publishing. So I would make these mandatory and the GitHub account optional.

bianchini88 commented:
> GitHub “issues”

Do we really need the ""? Also, the "I" in issues is often capitalised on web pages.

bianchini88 commented:
Other than the few minor points I raised above, I am happy with this PR, @bedroesb.

bedroesb commented:
> At the top of [this page](https://elixir-belgium.github.io/rdmkit-sandbox/how_to_contribute) what do we mean exactly with
> 
> > Provide acknowledgement for all your contributions.
> 
> ?

What do you think of:

* Acknowledge [contributions](#acknowledgement-and-ownership-of-content) of yourself and other contributors.

bedroesb commented:
> > If you want to link your GitHub account, ORCID or email address,
> 
> We have never discussed whether adding this information should be mandatory or not. I would argue that all people have an institutional email address and most people have an ORCID, as this information is also required for e.g. publishing. So I would make these mandatory and the GitHub account optional.

I will change the phrasing, but it will still be a "soft" requirement.

bedroesb commented:
> > GitHub “issues”
> 
> ([here](https://elixir-belgium.github.io/rdmkit-sandbox/github_way)) Do we really need the ""? Also, the "I" in issues is often capitalised on web pages.

Well spotted, I've updated it!

bedroesb commented:
@bianchini88 I think I've tackled your comments, let me know what you think

bianchini88 commented:
@bedroesb Now that I am looking at this again, I have another comment
In the section:

> If you wish to contribute, please remember the following for better consistency and efficiency:

2 out of 3 points have nothing to do with "consistency and efficiency". You can maybe just get rid of that and rephrase it. 

PS I am on leave this week, so expect slow responses.

bedroesb commented:
@bianchini88 No problem, I trimmed the sentence.
